lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -66,11 +66,19 @@ function dosomething_reportback_build_reportbacks_query($params = array()) {
   }
 
   if (isset($params['runs'])) {
-    if (is_array($params['runs'])) {
-      $query->condition('rb.run_nid', $params['runs'], 'IN');
-    } else {
-      $query->condition('rb.run_nid', $params['runs'], '=');
+    if (is_string($params['runs'])) {
+      $params['runs'] = explode(' ', $params['runs']);
     }
+
+    $campaign_runs = [];
+    
+    foreach ($params['runs'] as $campaign_run) {
+      if (preg_match('/^[0-9]+$/', $campaign_run)) {
```

Comment:
[`is_numeric`](http://php.net/manual/en/function.is-numeric.php) would make this a litlte more readable, although not sure what's exactly best practice / what we use the most -- cc @sergii-tkachenko ?